canopy/canopyjs/src/transactions/submitter.ts

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright 2019 Cargill Incorporated
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+interface BatchStatus {
+  statusType: string;
+  message: BatchMessage[];
+}
+
+interface BatchMessage {
+  transaction_id: string;
+  error_message: string;
+  error_data: number[];
+}
+
+interface BatchInfo {
+  id: string;
+  status: BatchStatus;
+}
+
+async function http(
```

Comment:
What is the benefit to writing our own http abstraction in place of using something like Fetch or Axios?

Reply:
We ran into some hard-to-debug issues using other libraries. E.g. Axios just doesn't handle octet-stream right at all.